### PR TITLE
stripe integration

### PR DESCRIPTION
**Pull request by Travmatth**
# Fixes
- check jwt access token is present on launch

## Proposed Changes
  - update component to accept payments
  - add stripe payment component into /profile
  - /stripe/:player middleware to verify player
  - /stripe/:player/intent to create new stripe payment intent
  - /stripe/webhook to listen for created payments and update player account
  - game logic to verify account permissions before launching new game

### CODE REVIEW

File: server/controllers/authenticate.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      return res.status(500).send({ error: `Error fetching user`});
+  }
+  res.locals.user = user;
+  next();
+};
+
+const playerHasPrivateGames = async (name) => {
+  try {
+    const user = await User.findOne({ name });
+    return user?.account.privateGames.enabled;
```
Comment (mpmpx) [inline server/controllers/authenticate.js]: Typo?

Comment (mpmpx) [reply server/controllers/authenticate.js]: Not a typo. Need to remove '?'

Comment (Travmatth) [reply server/controllers/authenticate.js]: [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) allows the line `user?.account.privateGames.enabled;` to return `undefined` if `user === null`, as opposed to throwing a type error when `(null).accont` is accessed. Our heroku is running on Node 14, the latest stable release which supports optional chaining

**Review by AltiV — COMMENTED** [review]
Seems to be missing stripe in both client and server package.jsons

**Comment by AltiV** [comment]
Your PR breaks being able to start public games, as the button results in an HTTP 500 Internal Server Error.

**Pull request merged**